share/spack/gitlab/cloud_pipelines/stacks/aws-pcluster-x86_64_v4/spack.yaml

```diff
@@ -69,131 +70,63 @@ spack:
-            - "cflags=-std=c18 target=x86_64_v3"
-          when: "%intel"
     gettext:
       # Newer gettext cannot build with gcc@12 and old AL2 glibc headers
       # Older gettext versions do not build correctly with oneapi.
       require:
-        - one_of:
-            - '@:0.20'
-            - '%oneapi'
+      - '@:0.20 %gcc'
     gromacs:
       require:
-        - one_of:
-            - "+intel_provided_gcc ^intel-oneapi-mkl target=x86_64_v4"
-            - "+intel_provided_gcc ^intel-oneapi-mkl target=x86_64_v3"
-          when: "%intel"
-        - one_of:
-            - "+intel_provided_gcc target=x86_64_v4 ^intel-oneapi-mkl"
-            - "+intel_provided_gcc target=x86_64_v3 ^intel-oneapi-mkl"
-          when: "%oneapi"
-    intel-oneapi-compilers:
-      require: "intel-oneapi-compilers %gcc target=x86_64_v3"
+      - +intel_provided_gcc ^intel-oneapi-mkl
     intel-oneapi-mpi:
       variants: +external-libfabric generic-names=True
     lammps:
       require:
-        - one_of:
-            - "lammps_sizes=bigbig +molecule +kspace +rigid +asphere +opt +openmp +openmp-package +intel fft=mkl ^intel-oneapi-mkl target=x86_64_v4"
-            - "lammps_sizes=bigbig +molecule +kspace +rigid +asphere +opt +openmp +openmp-package fft=mkl ^intel-oneapi-mkl target=x86_64_v3"
-          when: "%intel"
-        - one_of:
-            - "lammps_sizes=bigbig +molecule +kspace +rigid +asphere +opt +openmp +openmp-package +intel fft=mkl ^intel-oneapi-mkl target=x86_64_v4"
-            - "lammps_sizes=bigbig +molecule +kspace +rigid +asphere +opt +openmp +openmp-package fft=mkl ^intel-oneapi-mkl target=x86_64_v3"
-          when: "%oneapi"
-    libidn2:
-      require:
-        - one_of:
-            - "cflags=-std=c18 target=x86_64_v4"
-            - "cflags=-std=c18 target=x86_64_v3"
-          when: "%intel"
+      - lammps_sizes=bigbig +molecule +kspace +rigid +asphere +opt +openmp +openmp-package
+        fft=mkl ^intel-oneapi-mkl
     libfabric:
       buildable: true
       externals:
-        - prefix: /opt/amazon/efa/
-          spec: libfabric@1.17.0
-      require: ['fabrics=shm,efa']
-    libunistring:
+      - prefix: /opt/amazon/efa/
+        spec: libfabric@1.17.0
       require:
-        - one_of:
-            - "cflags=-std=c18 target=x86_64_v4"
-            - "cflags=-std=c18 target=x86_64_v3"
-          when: "%intel"
+      - fabrics=shm,efa
     mpas-model:
       require:
-        - one_of:
-            - "precision=single ^parallelio+pnetcdf target=x86_64_v4"
-            - "precision=single ^parallelio+pnetcdf target=x86_64_v3"
-          when: "%intel"
-        - one_of:
-            - "precision=single ^parallelio+pnetcdf target=x86_64_v4"
-            - "precision=single ^parallelio+pnetcdf target=x86_64_v3"
-          when: "%oneapi"
+      - one_of:
+        - precision=single ^parallelio+pnetcdf target=x86_64_v4
+        - precision=single ^parallelio+pnetcdf target=x86_64_v3
```

Comment:
This can actually be more simplified to just `require: precision=single ^parallelio+pnetcdf `